app_flutter/lib/commit_box.dart

```diff
@@ -5,29 +5,136 @@
+/// for the author of this commit. Clicking outside of the [OverlayEntry]
+/// will close it.
+class CommitBox extends StatefulWidget {
   // TODO(chillers): convert to use commit model
-  const CommitBox({
-    Key key,
-    @required this.message,
-    @required this.avatarUrl,
-  }) : super(key: key);
+  const CommitBox(
+      {Key key,
+      @required this.message,
+      @required this.avatarUrl,
+      @required this.author})
+      : super(key: key);
 
   /// Commit message that summarizes the change made.
   final String message;
 
   /// Image URL to the avatar of the author of this commit.
   final String avatarUrl;
 
+  /// The person that authored this commit.
+  final String author;
+
+  @override
+  _CommitBoxState createState() => _CommitBoxState();
+}
+
+class _CommitBoxState extends State<CommitBox> {
+  OverlayEntry _commitOverlay;
+
   @override
   Widget build(BuildContext context) {
-    // TODO(chillers): add overlay to view more information
-    return Container(
-      margin: const EdgeInsets.all(1.0),
-      child: Image.network(
-        avatarUrl,
-        height: 40,
+    return GestureDetector(
+      onTap: _handleTap,
+      child: Container(
+        margin: const EdgeInsets.all(1.0),
+        child: Image.network(
+          widget.avatarUrl,
+          height: 40,
+        ),
       ),
     );
   }
+
+  void _handleTap() {
+    _commitOverlay = OverlayEntry(
+        builder: (nullContext) => CommitOverlayContents(
+            parentContext: context,
+            widget: widget,
+            closeCallback: closeOverlay));
+
+    Overlay.of(context).insert(this._commitOverlay);
+  }
+
+  void closeOverlay() => _commitOverlay.remove();
+}
+
+class CommitOverlayContents extends StatelessWidget {
+  CommitOverlayContents({
+    Key key,
+    @required this.parentContext,
+    @required this.widget,
+    @required this.closeCallback,
+  }) : super(key: key);
+
+  /// The parent context that has the size of the whole screen
+  final BuildContext parentContext;
+
+  /// The parent widget that contains state variables
+  final CommitBox widget;
+
+  final void Function() closeCallback;
+
+  @override
+  Widget build(BuildContext context) {
+    RenderBox renderBox = parentContext.findRenderObject();
+
+    return Stack(
+      children: <Widget>[
+        // This is the area a user can click (the rest of the screen) to close the overlay.
+        GestureDetector(
+          onTap: closeCallback,
+          child: Container(
+            width: MediaQuery.of(parentContext).size.width,
+            height: MediaQuery.of(parentContext).size.height,
+            // Color must be defined otherwise the container can't be clicked on
+            color: Colors.transparent,
+          ),
+        ),
+        Positioned(
+          width: 300,
+          // Move this overlay to be where the parent is
+          top: renderBox.localToGlobal(Offset.zero).dy +
```

Comment:
I don't think that this operation is cached automatically, and it might be pretty expensive, it would be worth pulling out into a single computation:

```
final Offsetset topLeft = renderBox.localToGlobal(Offset.zero);
...
top: topLeft.dy + (renderBox.size.height / 2),
```

Reply:
That's a good point. I also made the render box variable final.